raven/src/main/java/com/getsentry/raven/marshaller/json/StackTraceInterfaceBinding.java

```diff
@@ -35,21 +35,21 @@ private void writeFrame(JsonGenerator generator, StackTraceElement stackTraceEle
         generator.writeStartObject();
         generator.writeStringField(FILENAME_PARAMETER, stackTraceElement.getFileName());
         generator.writeStringField(MODULE_PARAMETER, stackTraceElement.getClassName());
-        generator.writeBooleanField(IN_APP_PARAMETER, !(removeCommonFramesWithEnclosing && commonWithEnclosing)
-            && isFrameInApp(stackTraceElement));
+        boolean inApp = !(removeCommonFramesWithEnclosing && commonWithEnclosing) && isFrameInApp(stackTraceElement);
+        generator.writeBooleanField(IN_APP_PARAMETER, inApp);
         generator.writeStringField(FUNCTION_PARAMETER, stackTraceElement.getMethodName());
         generator.writeNumberField(LINE_NO_PARAMETER, stackTraceElement.getLineNumber());
         generator.writeEndObject();
     }
 
     private boolean isFrameInApp(StackTraceElement stackTraceElement) {
-        //TODO: A set is absolutely not efficient here, a Trie could be a better solution.
-        for (String notInAppFrame : notInAppFrames) {
-            if (stackTraceElement.getClassName().startsWith(notInAppFrame)) {
-                return false;
+        for (String inAppFrame : inAppFrames) {
```

Comment:
`// TODO: A linear search is absolutely not efficient here, a Trie could be a better solution.`